roles/edpm_nova/scripts/derive_pci_passthrough_whitelist.py

```diff
@@ -0,0 +1,587 @@
+        pf_path = os.path.join(_SYS_CLASS_NET_PATH, pf_name, "device")
+        vendor, vf_product = get_pci_device_info_by_ifname(pf_path, 'virtfn0')
+        if (('product_id' not in user_config or user_config['product_id'][-4:] == vf_product[-4:])
+                and pf_name in available_vfs):
+            for vf_addr in available_vfs[pf_name]:
+                user_address_regex = re.compile(user_address_pattern)
+                if user_address_regex.match(vf_addr):
+                    sel_addr.append(vf_addr)
+            if not sel_addr:
+                for vf_addr in allocated_pci:
+                    if user_address_regex.match(str(vf_addr)):
+                        """ When the user_config address matches that of the NIC Partitioned VF,
+                        the VF must be removed from the user_config. So return the status such
+                        that the caller ignores this user_config if this user_config is very
+                        specific to the NIC Partitioned VF. If the user_config resulted in the
+                        derivations of other configurations, the derived configuration shall
+                        replace the original user_config.
+                        """
+                        del_user_config = True
+
+    if sel_addr:
+        pci_passthrough = get_pci_passthrough_whitelist(
+            user_config, pf_name, sel_addr)
+    return pci_passthrough, del_user_config
+
+
+def get_passthrough_config_for_all_pf(user_config,
+                                      system_configs,
+                                      allocated_pci):
+    derived_config = []
+    nic_partition_pfs = get_sriov_nic_partition_pfs(system_configs)
+    non_nic_partition_pfs = get_sriov_non_nic_partition_pfs(system_configs)
+    del_user_config = False
+
+    """
+    For each user_config, do the matching for NIC Partitioning PFs/VFs
+    """
+    for pf in nic_partition_pfs:
+        passthrough_tmp, status = get_passthrough_config(
+            user_config, pf, allocated_pci, False)
+        del_user_config = del_user_config or status
+        if passthrough_tmp:
+            derived_config.extend(passthrough_tmp)
+
+    """
+    If there is no config added from NIC Part nics, then skip parsing the
+    NON NIC Partitioned PFs.
+    """
+    if (derived_config or del_user_config):
+        for pf in non_nic_partition_pfs:
+            passthrough_tmp, status = get_passthrough_config(
+                user_config, pf, allocated_pci, True)
+            derived_config.extend(passthrough_tmp)
+
+    if derived_config:
+        return derived_config, False
+    else:
+        return derived_config, del_user_config
+
+
+# Todo: (remove)This function will not be needed now
+def get_neutron_devmap_configs():
+    ml2_sriov = []
+    valid_spec = 'physical_device_mappings'
+
+    # Read phydev mapping from neutron conf file
+    try:
+        with open('/var/lib/os-net-config/01-neutron-sriov-agent.conf') as neutron_config:
```

Comment:
Since the final config map files for sriov-nova and sriov-neutron (for NFV) are not yet finalised in VA design, using the path which QE uses (i.e ci-framework file name)
This will be updated later once VA is updated (but updated plugin to take this file as user input)

Reply:
As per latest code:
https://github.com/openstack-k8s-operators/architecture/blob/c85878642bcd64eb67ad2ea19e4e6e8bdb790cfb/va/nfv/sriov/edpm/nova_sriov.yaml#L14